src/js/editor/editor.js

```diff
@@ -444,17 +444,26 @@ class Editor {
     const markerRenderNode = leftRenderNode;
     const marker = markerRenderNode.postNode;
     const section = marker.section;
-    const newMarkers = marker.split(leftOffset);
 
-    // FIXME rightMarker is not guaranteed to be there
-    let [leftMarker, rightMarker] = newMarkers;
+    let [beforeMarker, afterMarker] = marker.split(leftOffset);
+    if (!afterMarker) {
+      afterMarker = beforeMarker;
+      beforeMarker = null;
+    }
+
+    if (beforeMarker) {
+      section.markers.insertAfter(beforeMarker, marker);
+    } else {
+      // start of section, remove the existing marker and add a blank one
+      section.markers.remove(marker);
+      section.markers.append(this.builder.createBlankMarker());
```

Comment:
Yeah, agreed, this is fairly confusing. I'm reworking this to be a bit more declarative.